pkgs/development/python-modules/locust/default.nix

```diff
@@ -0,0 +1,194 @@
+{
+  lib,
+  buildPythonPackage,
+  pythonRelaxDepsHook,
+  python,
+  callPackage,
+  fetchFromGitHub,
+  poetry-core,
+  poetry-dynamic-versioning,
+  pytestCheckHook,
+  configargparse,
+  cryptography,
+  flask,
+  flask-cors,
+  flask-login,
+  gevent,
+  geventhttpclient,
+  msgpack,
+  psutil,
+  pyquery,
+  pyzmq,
+  requests,
+  retry,
+  tomli,
+  werkzeug,
+}:
+
+buildPythonPackage rec {
+  pname = "locust";
+  version = "2.31.2";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "locustio";
+    repo = "locust";
+    rev = "refs/tags/${version}";
+    hash = "sha256-xDquVQjkWVER9h0a6DHWRZH6KtRf0jsThycSojDEdh4=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail 'script = "pre_build.py"' ""
+
+    substituteInPlace locust/test/test_main.py \
+      --replace-fail '"locust"' '"${placeholder "out"}/bin/locust"'
+
+    substituteInPlace locust/test/test_log.py \
+      --replace-fail '"locust"' '"${placeholder "out"}/bin/locust"'
+  '';
+
+  webui = callPackage ./webui.nix {
+    inherit version;
+    src = "${src}/locust/webui";
+  };
+
+  preBuild = ''
+    mkdir -p $out/${python.sitePackages}/${pname}
+    ln -sf ${webui} $out/${python.sitePackages}/${pname}/webui
+  '';
+
+  build-system = [ poetry-core ];
+
+  nativeBuildInputs = [
+    poetry-dynamic-versioning
+    pythonRelaxDepsHook
+  ];
```

Comment:
`pythonRelaxDepsHook` is no longer needed. I suggest to move `poetry-dynamic-versioning` to `build-system`.

Reply:
Ah, good to know. Thanks :)